### PR TITLE
events: Update docs for beta release

### PR DESCRIPTION
**Pull request by swenson**

### CODE REVIEW

File: website/content/docs/concepts/events.mdx [hunk 1]
```diff
@@ -143,27 +138,56 @@ JSON for the requested events (one line per event):
...
+These checks are cached for a short period of time for performance reasons, so it is possible that
+events may be sent for a few minutes after a token has been revoked or a policy has been deleted.
 
 ## Supported versions
 
+In Vault 1.15+, events are enabled by default.
+
+For Vault 1.13 and 1.14, events are disabled by default when starting a Vault server and can be
+enabled via the `events.alpha1` [experiment option](/vault/docs/configuration#experiments) in the Vault
+configuration or on the command-line:
```
Comment (schavis) [inline website/content/docs/concepts/events.mdx:183]:
```suggestion

Version | Support
------- | -------
<= 1.12 | Not supported
1.13    | Supported; **disabled** by default
1.14    | Supported; **disabled** by default
1.15+   | Supported; **enabled** by default

For versions where events are disabled by default, you can enable the
functionality with the `events.alpha1`
[experiment option](/vault/docs/configuration#experiments) in your Vault
configuration or from the command line with the `-experiments` flag. For example:
```

Style correction: make key information easy to scan

File: website/content/docs/concepts/events.mdx [hunk 2]
```diff
@@ -143,27 +138,56 @@ JSON for the requested events (one line per event):
...
+$ vault server -experiment events.alpha1
+```
+
 | Vault Version | Support                                     |
 | ------------- | ------------------------------------------- |
 | <= 1.12       | Not supported                               |
-| 1.13          | Supported (with `events.alpha1` experiment) |
\ No newline at end of file
+| 1.13–1.14     | Supported (with `events.alpha1` experiment) |
+| 1.15+         | Supported                                   |
```
Comment (schavis) [inline website/content/docs/concepts/events.mdx:193]: Suggest moving the table to before the example command so it's clearer to folks whether or not the command example is relevant to them

File: website/content/docs/commands/events.mdx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+page_title: events - Command
+description: |-
+  The "events" command interacts with the Vault events subsystem.
+---
+
+# events
+
+The `events` command is used to subscribe to Vault
+[events](/vault/docs/concepts/events) via `events subscribe`. It shows a
+real-time display of events generated by Vault on standard out.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:12]:
```suggestion
Use the `events` command to get a real-time display of
[events](/vault/docs/concepts/events) generated by Vault and subscribe to Vault
events. Note that the `events subscribe` runs indefinitly and will not exit on
its own unless it encounters an unexpected error. Similar to `tail -f` in the
Unix world, you must terminate the process from the command line to end the
`events` command.
```

Style correction: write in active voice

File: website/content/docs/commands/events.mdx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+layout: docs
+page_title: events - Command
+description: |-
+  The "events" command interacts with the Vault events subsystem.
+---
+
+# events
+
+The `events` command is used to subscribe to Vault
+[events](/vault/docs/concepts/events) via `events subscribe`. It shows a
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:11]: Follow-up question: what does "it shows a real-time display of events" mean? Is there a separate `events` subcommand that lists available events or do we just return a list of active events as part of a successful subscription call?

File: website/content/docs/commands/events.mdx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+The desired event types (sometimes called "topics") must be specified, which
+may be a glob pattern (with `*` treated as a wildcard).
+
+The output will be JSON objects serialized using the default protobuf JSON
+serialization format, with one line per event received.
+
+Note that this command is designed to run indefinitely. It is similar to
+`tail -f` in the Unix world. This command will not exit on its own unless it
+encounters an unexpected error. As a user, you must terminate this process
+yourself to shut it down.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:23]:
```suggestion
```

Suggest including the behavioral note with the intro sentence so folks have a clear idea of how the command behaves right from the jump.

File: website/content/docs/commands/events.mdx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+The `events` command is used to subscribe to Vault
+[events](/vault/docs/concepts/events) via `events subscribe`. It shows a
+real-time display of events generated by Vault on standard out.
+
+The desired event types (sometimes called "topics") must be specified, which
+may be a glob pattern (with `*` treated as a wildcard).
+
+The output will be JSON objects serialized using the default protobuf JSON
+serialization format, with one line per event received.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:18]:
```suggestion
Specify the desired event types (also called "topics") as a glob pattern. To match
against multiple events, use `*` as a wildcard. The command returns serialized
JSON objects in the default protobuf JSON serialization format with one line per
event received.
```

Style correction: write in active voice

File: website/content/docs/commands/events.mdx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+Subscribe to all `kv-v2/data-write` events:
+
+```shell-session
+$ vault events subscribe kv-v2/data-write
+```
+
+Subscribe to all non-`data-write` KV events for the secret `secret/data/foo` in
+the current namespace as well as namespace `ns1`:
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:46]:
```suggestion
Subscribe to all KV events in the current and `ns1` namespaces for the secret `secret/data/foo` that do not involve writing data:
```

Suggest restructuring the sentence to make it a little easier to parse.

File: website/content/docs/commands/events.mdx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+```shell-session
+$ vault events subscribe -namespaces=ns1 -filter='data_path == secret/data/foo and operation != "data-write"' 'kv*'
+```
+
+## Usage
+
+The following flags are available in addition to the [standard set of
+flags](/vault/docs/commands) included on all commands.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:55]:
```suggestion
`events subscribe` supports the following flags in addition to the [standard set of
flags](/vault/docs/commands) included on all commands.
```

Style correction: write in active voice

File: website/content/docs/commands/events.mdx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+### Options
+
+- `-timeout`: `(duration: "")` - close the WebSocket automatically after the
+  specified duration.
+
+- `-filter` `(string: "")` - Filter expression used to select events to be sent
+  through the WebSocket.
+
+  See [Boundary's docs](/boundary/docs/concepts/filtering) on how these
+  expressions are evaluated and what options are available.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:66]: Why are we sending folks to the Boundary docset?

File: website/content/docs/commands/events.mdx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+	- `namespace`: the path of the namespace that created the event, e.g., `ns1/`
+
+  By default, the filter will be empty and will match all events that the
+  request has access to for the given event types specified in the request.
+
+  A filter could be as simple as `data_path == secret/data/foo` to specify that
+  only events that refer to the `data_path` of `secret/data/foo` should be sent
+  via the WebSocket.
+
+  A filter can also be more complex, for example:
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:83]:
```suggestion
The filter string is empty by default. Unfiltered subscription requests match to
all events that the requestor has access to for the target event type. When the
filter string is not empty, Vault applies the filter conditions after the policy
checks to narrow the events provided in the response.

Filters can be straightforward path matches like `data_path == secret/data/foo`,
which specifies that Vault should pass return events that refer to the
`secret/data/foo` secret to the WebSocket. Or more complex statements that
exclude specific operations. For example:
```
Style correction: write in active voice, avoid implying something is/isn't "simple"

File: website/content/docs/commands/events.mdx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  ```
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:84]: I think the code fence should be typed as `shell-session` here.

File: website/content/docs/commands/events.mdx [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+  data_path == secret/data/foo and operation != write
+  ```
+
+  The filter is applied after policy checks.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:88]:
```suggestion
```

Folded into the suggestion above

File: website/content/docs/commands/events.mdx [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+### Enterprise Options
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:90]:
```suggestion
### Enterprise options
```

Style correction: use sentence case for headings

File: website/content/docs/commands/events.mdx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+  automatically prepended, so specifying `ns2` when the request is in the `ns1`
+  namespace will result in subscribing to `ns1/ns2` (in addition to `ns1`).
+  Patterns can include `*` characters to indicate wildcards. The default is to
+  subscribe only to the request's namespace.
+
+  -> Note: A root token, or a token associated with entities and groups with
+     the appropriate policies, must be used for the event subscription
+     request to be able to subscribe to multiple namespaces. See
+     [this tutorial](/vault/tutorials/enterprise/namespaces) for an example
+     setup.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:105]:
```suggestion
<Note>

  To subscribe to events across multiple namespaces, you must provide a root
  token or a token associated with appropriate policies across all the targeted
  namespaces. Refer to the
  [Secure multi-tenancy with namespaces](/vault/tutorials/enterprise/namespaces)
  tutorial for configuring your Vault instance appropriately.

</Note>
```

Style correction: use new notation for asides, write in active voice, avoid generic anchor text

File: website/content/docs/commands/events.mdx [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+### Enterprise Options
+
+<EnterpriseAlert product="vault" />
+
+- `-namespaces` `(string)` - Additional child namespaces to subscribe to. Can
+  be repeated to add more namespace patterns. The namespace of the request is
+  automatically prepended, so specifying `ns2` when the request is in the `ns1`
+  namespace will result in subscribing to `ns1/ns2` (in addition to `ns1`).
+  Patterns can include `*` characters to indicate wildcards. The default is to
+  subscribe only to the request's namespace.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:99]:
```suggestion
- `-namespaces` `(string)` - Additional **child** namespaces for the
  subscription. Repeat the flag to add additional namespace patterns to the
  subscription request. Vault automatically prepends the issuing namespace for
  the request to the provided namespace. For example, if you include
  `-namespaces=ns2` on a request made in the `ns1` namespace, Vault will attempt
  to subscribe you to events under the `ns1/ns2` and `ns1` namespaces. You can
  use the `*` character to include wildcards in the namespace pattern. By
  default, Vault will only subscribe to events in the requesting namespace.
```

Style correction: write in active voice with one idea per sentence

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/commands/events.mdx (hunk 4 above)
Comment (swenson) [reply website/content/docs/commands/events.mdx]: There is no subcommand for showing available events. This returns an on-going list of all events that are happening, and will continue forever (as opposed to almost every other Vault command, which do one thing, return a response, and are done).

File: website/content/docs/commands/events.mdx (hunk 9 above)
Comment (swenson) [reply website/content/docs/commands/events.mdx:67]: That page is independent of Boundary. They just happen to have written the best documentation on the language implemented by hashicorp/go-bexpr, and there's no sense duplicating it.

File: website/content/docs/commands/events.mdx (hunk 11 above)
Comment (swenson) [reply website/content/docs/commands/events.mdx:86]: It's not a shell-session though. It's a generic boolean expression in the hashicorp/go-bexpr language.

File: website/content/docs/concepts/events.mdx [hunk 16]
```diff
@@ -143,27 +138,56 @@ JSON for the requested events (one line per event):
...
+
+   ```hcl
+   path "secret/my-data" {
+     capabilities = ["list", "subscribe"]
+     subscribe_event_types = ["*"]
+   }
+   ```
+
+-> Note: if the policy for the *path of the secret* also contains the `read` capability, we **may** include
+   the contents of the secret in the event payload in a future version of Vault.
```
[the remark this answers is not in the document]
Comment (swenson) [reply website/content/docs/concepts/events.mdx]: The `Highlight` stops all Markdown formatting from occurring. I like the way the `Highlight` looks though. But, a downside is that it is not possible to replicate the full backtick formatting (as far as I can tell) due to the way mdx processes the markdown.

**Comment by swenson** [comment]
Thanks @schavis! I integrated your suggestions and did a little formatting cleanup after applying them.

File: website/content/docs/commands/events.mdx [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+### Options
+
+- `-timeout`: `(duration: "")` - close the WebSocket automatically after the
+  specified duration.
+
+- `-filter` `(string: "")` - Filter expression used to select events to be sent
+  through the WebSocket.
+
+  See [Boundary's docs](/boundary/docs/concepts/filtering) on how these
+  expressions are evaluated and what options are available.
```
Comment (schavis) [inline website/content/docs/commands/events.mdx:62]:
```suggestion
Refer to the [Filter expressions](/boundary/docs/concepts/filtering) guide
in the Boundary documentation for a complete list of filtering options and an
explanation on how we evaluate filter expressions.
```

Style correction: avoid general anchor text

**Review by schavis — CHANGES_REQUESTED** [review]
Added one more suggestion to clarify the link to Boundary docs. Otherwise, looks good.

**Review by schavis — APPROVED** [review]

**Comment by swenson** [comment]
Thanks!

**Pull request merged**